aspnetcore/mvc/advanced/custom-model-binding/3.0sample/PolymorphicModelBinding/Pages/AddDevice.cshtml

```diff
@@ -0,0 +1,32 @@
+﻿@page
+@model AddDeviceModel
+@{
+    ViewData["Title"] = "Add a device";
+}
+
+<h1>@ViewData["Title"]</h1>
+
+<div class="row">
+    <div class="col-md-4">
+        <form method="POST">
+            <div asp-validation-summary="All"></div>
+
+            <div class="form-group">
+                <label for="Device_Kind">Kind</label>
+                <select id="Device_Kind" name="Device.Kind" asp-items="Model.DeviceKinds" class="form-control"></select>
+            </div>
+
+            <div class="form-group">
+                <label for="Device_CPUIndex">CPUIndex</label>
```

Comment:
We can't use `asp-for` because we're binding against `Device`, the base class, which doesn't have properties for `CPUIndex` and `ScreenSize`. We could use `asp-for` for the `Kind` property, but that might be confusing mixed in with the others.

There's also no validation on these. If you choose a `Kind` and don't fill out `CPUIndex` or `ScreenSize` accordingly, we just show it as empty in the bootstrap alert. Handling all of that might be overkill for the sample, so I left it. I was tempted to add some js to show/hide the elements, but, again, the sample thing. :)